automation/sharded_repeated_test.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+function usage {
+    cat <<EOF
+usage: $0 [-h|--help] show help
+       $0 [num_shards [shard_index]]
+
+       runs a shard of the e2e tests as in 'repeated_test.sh'. Shard to run is automatically determined by time of day. Default is four shards, where
+       the first shard would be run from 0-5AM, 2nd would be run from 6-11AM ....
+
+       Note: this script is calling 'repeated_test.sh' to run the tests.
+EOF
+}
+
+function find_tests {
+    find tests/ -name '*.go' | grep -vE '^tests/(assert|framework|utils.go|lib|performance|realtime)|_suite(_test)\.go'
+}
+
+function main {
+    if [[ $# -gt 0 ]] && [[ $1 =~ -h ]]; then
+        usage
+        exit 0
+    fi
+
+    set -x
+
+    num_shards=${1-4}
+    hours_per_shard=$(( 24 / num_shards ))
+    if [[ $# -lt 2 ]]; then
+        shard_index=$(( $(date +%H) / hours_per_shard ))
+    else
+        shard_index=${2}
+        if [[ $shard_index -ge $num_shards ]]; then
+            echo "num_shards=$num_shards <= $shard_index!"
+            exit 1
+        fi
+    fi
+
+    num_lines=$(find_tests | wc -l)
+    if [[ ${num_lines} -eq 0 ]]; then
+        echo "No tests"
+        exit 0
+    fi
+    num_lines_per_segment=$(( num_lines / num_shards ))
+    start_line=$(( shard_index * num_lines_per_segment ))
+    end_line=$(( (shard_index + 1) * num_lines_per_segment ))
```

Comment:
what would happen here if `num_shards` did not divide cleanly into `num_lines` ? 

would this effect the `start_line` and `end_line` ?

Reply:
TBH I am no expert in using double parentheses, but I was remembering that the double parentheses would use integer for evaluation.

The `bash` man page says:
```
... Evaluation is done in fixed-width  integers
       with no check for overflow, though division by 0 is trapped and flagged as an
       error.
...
```

So I understood it will still stay in integer values. I assume that rounding doesn't matter that much here.

WDYT?